crates/storage/src/pack.rs

```diff
@@ -79,6 +126,23 @@ impl<T> Pack<T> {
     pub fn as_inner_mut(pack: &mut Pack<T>) -> &mut T {
         &mut pack.inner
     }
+
+    /// Sets the key.
+    fn set_key(mut self, value: Option<Key>) -> Self {
+        self.key = value;
+        self
+    }
```

Comment:
Why would you need this super low-level and fragile API?

Reply:
I switched to an alternative constructor now:
```
-        Pack::from(forward_pull_packed::<T>(ptr)).set_key(Some(ptr.key()))
+        let inner = forward_pull_packed::<T>(ptr);
+        Pack::new_with_key(inner, *ptr.key())
```
I mean, it's still an API function now, but at least the object can't be modified in hindsight.